### ISSUE TITLE
Sass support

### ISSUE THREAD

**Issue by fc1943s** [zanaptak/TypedCssClasses#4]
Would it be possible to support Sass/SCSS parsing? \o/

```
.outer1 .inner1 {
  // works
}
.outer2 {
  .inner2 {
    // nop. only outer2
  }
}
```

**Comment by zanaptak** [comment]
I will have to consider it. I don't think I want to have a hard dependency on a sass compiler internally due to complexity/maintenance. But maybe some way to call out to a user-configurable external process to compile the CSS could work. Can't guarantee anything or commit to any timeframe right now but I'll keep it in mind.

As a workaround you can try having a build script produce a generated CSS artifact from the source SCSS, and use the generated file for the type provider. For example in [this sample](https://github.com/zanaptak/TypedCssClasses/tree/master/sample/FableTailwind) I use a script with a call to `tailwind build` to produce a suitable generated reference CSS file, you could try something similar with a `sass` command.

**Comment by fc1943s** [comment]
For now I'm keeping a copy of the nested classes at the root level with an empty selector, so the library can detect it. Having the static F# code is the main goal, and it works beautifully.

I had suggested it without looking deeply at how the library parses the CSS, imagining it would be some kind of regex selector. A sass compiler didn't even cross my mind, but it could be a less-hacky option to not implement the whole parsing by hand. There is the complexity and performance issues tho.

**Comment by zanaptak** [comment]
A mechanism to call external commands is now implemented in 0.3.0 and will be the recommended solution for this (as opposed to direct internal parse support). See main page readme for details.
